### ISSUE TITLE
Problems

### ISSUE THREAD

**Issue by NaitorStudios** [Rhadamus/CTF-Perspective#2]
It has come to my attention that this is a decompilation of private Clickteam work.
Unfortunately this can get users banned.
Perspective object for Windows is open source (and you can get source for most extensions if you ask Clickteam nicely), so you could have really ported it instead of stealing from Clickteam...

You can find the source code here:
https://github.com/Andos/Perspective

I'm afraid users of this decompiled extension might be tracked down, be banned and have their products keys revoked.

I suggest that you rewrite it yourself with the source I provided above if you really care about Clickteam and the community.

I simply don't want to see people getting hurt by this situation and I hope that you understand.

I'm not affiliated to Clickteam, I'm just trying to help.

**Comment by Rhadamus** [comment]
I've made so many edits to the source code that even if I were to rewrite this by myself using just the Windows code as a reference, I'd probably just end up with a nearly exact copy of the code I have right now. The main function that makes this thing tick was completely broken in the decompilation so I had to basically rewrite it from scratch. So I don't think there's a point in rewriting the code now.

**Comment by NaitorStudios** [comment]
Well, unfortunately I don't make the rules, so being hybrid or not, this will leave a stain that you could have been avoided...

**Comment by AITYunivers** [comment]
Classic Cesar, empty threats, ignore him lmfao.

**Comment by AITYunivers** [comment]
Riddle me this, Batman. If Clickteam cared about their userbase, why would they keep this from the community in the first place? Why did it have to be decompiled? Why should we care about a company that doesn't care in return?

**Comment by 1987kostya** [comment]
Yuni sent this to me with, probably, serious thoughts, but I'm laughing so hard that I don't even know what to say. You've done a great job with that extension, Rhadamus. Wish you all the best!

**Comment by AITYunivers** [comment]
I said "Riddle me this, Batman" what makes you think I'm being serious? lmfao

**Comment by NaitorStudios** [comment]
> If Clickteam cared about their userbase, why would they keep this from the community in the first place? Why did it have to be decompiled? Why should we care about a company that doesn't care in return?

They haven't shared it because it was not complete.

**Comment by AITYunivers** [comment]
Didn't stop Rhadamus, now did it?

**Comment by 1987kostya** [comment]
yeah, you're using that excuse everytime anyone releases anything clickteam tried to hide
